lib/utils/payment_hash_poller.dart

```diff
@@ -0,0 +1,58 @@
+// ignore_for_file: avoid_print
+import 'dart:async';
+
+import 'package:breez_sdk/bridge_generated.dart';
+import 'package:c_breez/services/injector.dart';
+
+class PaymentHashPoller {
+  final String paymentHash;
+  void Function(bool contains) onDone;
+
+  PaymentHashPoller({
+    required this.onDone,
+    required this.paymentHash,
+  });
+
+  void start() {
+    print("Start polling for payment hash: $paymentHash every 10 seconds");
+    final injector = ServiceInjector();
+    final breezLib = injector.breezLib;
+
+    Timer? paymentReceivedTimer;
+
+    paymentReceivedTimer = Timer.periodic(
+      const Duration(seconds: 10),
```

Comment:
Perhaps every 3-5 seconds is better here. It is a cheap call and we better know as soon as the payment is competed.

Reply:
I agree.
I'm also thinking of adding type & time filtering to listPayments call to reduce the list size. Which should reduce time complexity of the lookup on worst case to O(n) where n is most likely to be 1.

Reply:
Done 16849692e1a649732e9cb85ed63ad85e91fb13c0
We check the payment_hash against the received payments in the last 30 minutes.